File: java/domain/src/main/java/com/youknow/domain/repository/MoviesRepository.java
package com.youknow.domain.repository;

import com.youknow.domain.model.Movie;
import com.youknow.domain.model.SimpleMovie;

import java.util.List;

public interface MoviesRepository {

    interface MoviesLoadedCallback {
        void onMoviesLoaded(List<SimpleMovie> movies);
    }

    interface MovieLoadedCallback {
        void onMovieLoaded(Movie movie);
    }

    void getNowPlaying(MoviesLoadedCallback callback);

    void getUpcoming(MoviesLoadedCallback callback);

    void getMovie(String movieId, MovieLoadedCallback callback);

}

File: kotlin-databinding-coroutine-aac/app/src/main/java/com/youknow/movie/domain/repository/MoviesRepository.kt
package com.youknow.movie.domain.repository

import com.youknow.movie.domain.model.Movie
import com.youknow.movie.domain.model.SimpleMovie

interface MoviesRepository {

    suspend fun getNowPlaying(): List<SimpleMovie>

    suspend fun getUpcoming(): List<SimpleMovie>

    suspend fun getMovie(id: String): Movie
}